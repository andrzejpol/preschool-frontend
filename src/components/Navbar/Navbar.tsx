import {faMagnifyingGlass, faMessage, faUserNinja} from "@fortawesome/free-solid-svg-icons";
import {FontAwesomeIcon} from "@fortawesome/react-fontawesome";
import {useSelector} from "react-redux";

const Navbar = () => {

  const {userFirstName, userRole} = useSelector(state => state.userData);


  return (
    <div className="flex items-center justify-between p-4">
      <div
        className="hidden md:flex flex items-center gap-2 text-xs rounded-full ring-[1.5px] ring-gray-300 px-2">
        <FontAwesomeIcon icon={faMagnifyingGlass} className="flex items-center justify-center"/>
        <input type="text" placeholder="Search..." className="w-[200px] p-2 bg-transparent outline-none"/>
      </div>
      <div className="flex items-center gap-6 justify-end w-full">
        <div className="bg-white rounded-full w-7 h-7 flex items-center justify-between cursor-pointer relative">
          <FontAwesomeIcon icon={faMessage} className="flex items-center justify-center"/>
          <div
            className="absolute -top-2 -right-1 w-5 h-5 flex items-center justify-center bg-purple-500 text-white rounded-full text-xs">1
          </div>
        </div>
        <div className="flex flex-col">
          <span className="text-sm leading-3 font-medium">Witaj, {userFirstName}</span>
          <span className="text-[10px] text-gray-500 text-right">{userRole}</span>
        </div>
        <FontAwesomeIcon icon={faUserNinja} className="flex items-center justify-center w-[30px] h-[30px]"/>
      </div>
    </div>
  )
};

export default Navbar;