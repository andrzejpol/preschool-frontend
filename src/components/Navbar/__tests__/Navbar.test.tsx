import React from "react";
import {render} from "@testing-library/react";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import Navbar from "../Navbar";

const mockStore = configureStore([]);

describe("Navbar Component", () => {
  let store;

  beforeEach(() => {
    store = mockStore({
      userData: {
        userFirstName: "John",
        userRole: "Admin",
      },
    });
  });

  it("matches the snapshot", () => {
    const {asFragment} = render(
      <Provider store={store}>
        <Navbar/>
      </Provider>
    );

    // Snapshot assertion
    expect(asFragment()).toMatchSnapshot();
  });
});
