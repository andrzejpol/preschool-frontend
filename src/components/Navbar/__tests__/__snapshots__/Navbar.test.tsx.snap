// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Navbar Component > matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex items-center justify-between p-4"
  >
    <div
      class="hidden md:flex flex items-center gap-2 text-xs rounded-full ring-[1.5px] ring-gray-300 px-2"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-magnifying-glass flex items-center justify-center"
        data-icon="magnifying-glass"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352a144 144 0 1 0 0-288 144 144 0 1 0 0 288z"
          fill="currentColor"
        />
      </svg>
      <input
        class="w-[200px] p-2 bg-transparent outline-none"
        placeholder="Search..."
        type="text"
      />
    </div>
    <div
      class="flex items-center gap-6 justify-end w-full"
    >
      <div
        class="bg-white rounded-full w-7 h-7 flex items-center justify-between cursor-pointer relative"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-message flex items-center justify-center"
          data-icon="message"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M64 0C28.7 0 0 28.7 0 64L0 352c0 35.3 28.7 64 64 64l96 0 0 80c0 6.1 3.4 11.6 8.8 14.3s11.9 2.1 16.8-1.5L309.3 416 448 416c35.3 0 64-28.7 64-64l0-288c0-35.3-28.7-64-64-64L64 0z"
            fill="currentColor"
          />
        </svg>
        <div
          class="absolute -top-2 -right-1 w-5 h-5 flex items-center justify-center bg-purple-500 text-white rounded-full text-xs"
        >
          1
        </div>
      </div>
      <div
        class="flex flex-col"
      >
        <span
          class="text-sm leading-3 font-medium"
        >
          Witaj, John
        </span>
        <span
          class="text-[10px] text-gray-500 text-right"
        >
          Admin
        </span>
      </div>
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-user-ninja flex items-center justify-center w-[30px] h-[30px]"
        data-icon="user-ninja"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M224 256c-57.2 0-105.6-37.5-122-89.3c-1.1 1.3-2.2 2.6-3.5 3.8c-15.8 15.8-38.8 20.7-53.6 22.1c-8.1 .8-14.6-5.7-13.8-13.8c1.4-14.7 6.3-37.8 22.1-53.6c5.8-5.8 12.6-10.1 19.6-13.4c-7-3.2-13.8-7.6-19.6-13.4C37.4 82.7 32.6 59.7 31.1 44.9c-.8-8.1 5.7-14.6 13.8-13.8c14.7 1.4 37.8 6.3 53.6 22.1c4.8 4.8 8.7 10.4 11.7 16.1C131.4 28.2 174.4 0 224 0c70.7 0 128 57.3 128 128s-57.3 128-128 128zM0 482.3C0 399.5 56.4 330 132.8 309.9c6-1.6 12.2 .9 15.9 5.8l62.5 83.3c6.4 8.5 19.2 8.5 25.6 0l62.5-83.3c3.7-4.9 9.9-7.4 15.9-5.8C391.6 330 448 399.5 448 482.3c0 16.4-13.3 29.7-29.7 29.7L29.7 512C13.3 512 0 498.7 0 482.3zM160 96c-8.8 0-16 7.2-16 16s7.2 16 16 16l128 0c8.8 0 16-7.2 16-16s-7.2-16-16-16L160 96z"
          fill="currentColor"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;
